packaging/Makefile

```diff
@@ -68,65 +64,88 @@ docker.build.enable:
 	GOARCH=arm \
 	GOARM=7 \
 	CGO_ENABLED=$(CGO_ENABLED) \
-	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o build/linux/arm/v7/$(notdir $@) ../cmd/$(notdir $@)/*.go
+	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o $@ ../cmd/$(notdir $@)
 
 build/darwin/amd64/%:
-	GO111MODULE=$(GO111) \
 	GOOS=darwin \
 	GOARCH=amd64 \
 	CGO_ENABLED=$(CGO_ENABLED) \
-	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o build/darwin/amd64/$(notdir $@) ../cmd/$(notdir $@)/*.go
+	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o $@ ../cmd/$(notdir $@)
 
 build/darwin/arm64/%:
-	GO111MODULE=$(GO111) \
 	GOOS=darwin \
 	GOARCH=arm64 \
 	CGO_ENABLED=$(CGO_ENABLED) \
-	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o build/darwin/arm64/$(notdir $@) ../cmd/$(notdir $@)/*.go
+	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o $@ ../cmd/$(notdir $@)
 
 build/windows/amd64/%:
-	GO111MODULE=$(GO111) \
 	GOOS=windows \
 	GOARCH=amd64 \
 	CGO_ENABLED=$(CGO_ENABLED) \
-	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o build/windows/amd64/$(notdir $@) ../cmd/$(notdir $@)/*.go
+	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o $@ ../cmd/$(notdir $@)
 
-build.package: build.linux build.darwin build.windows
+.PHONY: build.package
+build.package: build.linux build.darwin build.windows ## package binaries
 	tar --transform 's,^\.,skipper-$(VERSION)-linux-amd64,' -C build/linux/amd64 -czvf skipper-$(VERSION)-linux-amd64.tar.gz .
 	tar --transform 's,^\.,skipper-$(VERSION)-linux-arm64,' -C build/linux/arm64 -czvf skipper-$(VERSION)-linux-arm64.tar.gz .
 	tar --transform 's,^\.,skipper-$(VERSION)-linux-armv7,' -C build/linux/arm/v7 -czvf skipper-$(VERSION)-linux-armv7.tar.gz .
 	tar --transform 's,^\.,skipper-$(VERSION)-darwin-amd64,' -C build/darwin/amd64 -czvf skipper-$(VERSION)-darwin-amd64.tar.gz .
 	tar --transform 's,^\.,skipper-$(VERSION)-darwin-arm64,' -C build/darwin/arm64 -czvf skipper-$(VERSION)-darwin-arm64.tar.gz .
 	tar --transform 's,^\.,skipper-$(VERSION)-windows-amd64,' -C build/windows/amd64 -czvf skipper-$(VERSION)-windows-amd64.tar.gz .
 	for f in *.tar.gz; do sha256sum $$f >> sha256sum.txt; done
+
+.PHONY: test
+test: ## test build targets using https://github.com/multiarch/qemu-user-static
+	IMAGE=skipper-packaging-test:local make docker.build.local
+	IMAGE=skipper-packaging-test:local make test-run
+
+	IMAGE=skipper-packaging-test:amd64 make docker.build.amd64
+	IMAGE=skipper-packaging-test:amd64 TARGETPLATFORM=linux/amd64 make test-run
+
+	ARM64_IMAGE=skipper-packaging-test:arm64 make docker.build.arm64
+	IMAGE=skipper-packaging-test:arm64 TARGETPLATFORM=linux/arm64 make test-run
+
+	ARM_IMAGE=skipper-packaging-test:armv7 make docker.build.armv7
+	IMAGE=skipper-packaging-test:armv7 TARGETPLATFORM=linux/arm/v7 make test-run
```

Comment:
Should/Can we also test the multi-arch image? Maybe even by running it on different (emulated) archs?

Reply:
I omitted multiarch here because it is not possible to build it without pushing. I guess to test it we'd need to push it first.